### ISSUE TITLE
Unexpected no-unused-vars behavior.

### ISSUE THREAD

**Issue by afahim** [eslint/eslint#5317]
**What version are you using?**
2.1.0

**What did you do?**
Ran `no-unused-vars` rule against unused `requirejs` imports.

```javascript
define["backbone", "underscore", "moment"],
function(backbone, underscore, moment) {
    console.log(moment);
}
``` 

**What happened?**
ESLint didn't throw an error.

**What did you expect to happen?**
I expected ESlint to throw an error, since two variables here, `backbone` and `underscore` aren't being used.

Let me know if this error is actually not within the scope of this rule.

I've also written up two failing test cases for no-unused-vars which fail, for more context, along with the test runner output. 

```javascript
{
    code:"define([\"backbone\", \"underscore\", \"moment\"], function(backbone, underscore, moment) {console.log(moment)});",
    errors: [
        {message: "'backbone' is defined but never used"},
        {message: "'underscore' is defined but never used"}
    ]
},
```
Output

```
AssertionError: Should have 2 errors but had 0: []
```

```javascript
{
    code:"define([\"backbone\", \"underscore\"], function(backbone, underscore) {});",
    errors: [
        {message: "'backbone' is defined but never used"},
        {message: "underscore' is defined but never used"}
    ]
}
```

Output

```
AssertionError: Should have 2 errors but had 1: [ { ruleId: 'no-unused-vars',
severity: 1,
message: '\'underscore\' is defined but never used',
line: 1,
column: 55,
nodeType: 'Identifier',
source: 'define(["backbone", "underscore"], function(backbone, underscore) {});' } ]
```

**Comment by afahim** [comment]
Oh I see, just learned about after-args. Never mind.
